packages/valory/skills/decision_maker_abci/behaviours/blacklisting.py

```diff
@@ -45,9 +46,13 @@ def _blacklist(self) -> None:
         """Blacklist the sampled bet."""
         sampled_bet_index = self.synchronized_data.sampled_bet_index
         sampled_bet = self.bets[sampled_bet_index]
+
         # the question is blacklisted, i.e., we did not place a bet on it,
         # therefore, we decrease the number of bets which was increased on sampling
-        sampled_bet.n_bets -= 1
+        if sampled_bet.queue_status == QueueStatus.TO_PROCESS:
+            sampled_bet.queue_status = QueueStatus.PROCESSED
+        elif sampled_bet.queue_status == QueueStatus.PROCESSED:
+            sampled_bet.queue_status = QueueStatus.REPROCESSED
```

Comment:
Let's extract a method for this and reuse it here and in `update_bet_transaction_information`.